task.go

```diff
@@ -113,7 +114,7 @@ type CheckpointTaskInfo struct {
 	// ParentCheckpoint is the digest of a parent checkpoint
 	ParentCheckpoint digest.Digest
 	// Options hold runtime specific settings for checkpointing a task
-	Options interface{}
+	Options *runctypes.CheckpointOptions
```

Comment:
can we keep this generic?

Reply:
Yes, we can. @kolyshkin and I could not understand this trick. Could you explain its idea?

Reply:
@crosbymichael from my perspective having `interface{}` here makes the rest of the code unnecessary complicated (with extra casts and type checks). Can you elaborate on why you want this to be of `interface{}` type? I could (theoretically) anticipate a C/R engine other than CRIU but practically this is too far in the future I'd care for now.

Reply:
https://github.com/google/gvisor (runsc) also has checkpoint/restore capability, and we are actively working on hooking it up to docker (i.e. fixing moby and stuff) as well. See https://github.com/moby/moby/pull/37360.

So there already is another C/R engine.